### ISSUE TITLE
not all operational daemonsets are scheduled on the tainted nodes

### ISSUE THREAD

**Issue by Ottovsky** [kubermatic/kubermatic#7142]
**User Story**
I added a tainted nodepool to the user cluster (tunneling expose strategy), but noticed that no pod could run there. Even though the pods were scheduled, they were failing during setup of the container runtime. The reason for this was canal pods couldn't establish connection to the apiserver and were crash looping. It turned out, they can't connect to the apiserver because the envoy-agent is missing in the tainted nodes, envoy-agent daemonset is missing tolerations for the taints. As consequence the tainted nodes are unusable:

```
kubectl get ds canal -o go-template='{{ .spec.template.spec.tolerations }}'
[map[effect:NoSchedule operator:Exists] map[key:CriticalAddonsOnly operator:Exists] map[effect:NoExecute operator:Exists]]
```
vs. 
```
kubectl get ds envoy-agent -o go-template='{{ .spec.template.spec.tolerations }}'
<no value>
```

**Acceptance criteria**
All the required operational daemonsets are able to schedule pods on the tainted nodes.

**Comment by kirek007** [comment]
Probably related to https://github.com/kubermatic/kubermatic/issues/7125

**Comment by redradrat** [comment]
This essentially means one cannot use tainted NoSchedule node pools, which effectively prohibits node segmentation on user clusters in general.

All the required operational daemonsets should ignore taints, right?

**Comment by s1gs3gv** [comment]
In our case the **node-local-dns** is not scheduled on the tainted node which makes all connections from the pod to the outside world unresolvable; incapacitated our app which requires such connectivity. Upon further inspection I saw that the canal pods are scheduled just fine and that they have these two tolerations which help them match any taints: 
```
:NoSchedule op=Exists
:NoExecute op=Exists 
```
I was wondering if perhaps this is possible path to solution to add these tolerations:

https://github.com/kubermatic/kubermatic/blob/43d0e7bbaa8c445ca9dd5931a3466e40ebf199a4/pkg/controller/user-cluster-controller-manager/resources/resources/node-local-dns/deamonset.go#L62

Of course this is not the only operational/system deamonset that is affected (user-ssh-keys-agent, logrotate)

**Comment by hdurand0710** [comment]
Closed as tested in #8659
